src/front/zsharpcurly/interp.rs

```diff
@@ -45,5 +45,10 @@ pub fn extract(
                 })
                 .collect::<Result<Vec<_>, _>>()?,
         )),
+        Ty::Tuple(tys) => Ok(T::new_tuple(
+            tys.iter()
+                .map(|ty| extract(name, ty, scalar_input_values))
```

Comment:
I would expect the `name` argument to the recursive call to be different for each iteration. Why is that not the case?

Reply:
Nice catch, tuples were not implemented correctly. I updated the code in a few positions and added test cases. Here, we do something similar to arrays.